projects/storefrontlib/shared/components/captcha/captcha.model.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Observable } from 'rxjs';
+import { CaptchaConfig } from '@spartacus/core';
+
+export interface RenderParams {
+  element?: HTMLElement | string;
+  publicKey?: string;
+}
+
+export interface CaptchaProvider {
+  /**
+   * Renders captcha widget
+   *
+   * @params parameters required by a provider
+   * @returns Observable for the key returned by a provider once user validated
+   */
+  renderCaptcha(params: RenderParams): Observable<string>;
```

Comment:
Should rendering really be part of a model?

Reply:
yep, maybe a new abstract class for  CaptchaProvider is better